ui/src/app/shared/modal/save-as-code/update.as.code.component.ts

```diff
@@ -76,63 +82,39 @@ export class UpdateAsCodeComponent {
                     });
-                    this.webworkerSub = webworker.response().subscribe(operation => {
-                        if (operation) {
-                            zone.run(() => {
-                                this.ope = JSON.parse(operation);
-                                if (this.ope.status > 1) {
-                                    this.loading = false;
-                                    webworker.stop();
-                                }
-                                this._cd.markForCheck();
-                            });
-                        }
-                    });
-                });
                 break;
             case 'pipeline':
-                this._pipService
-                    .updateAsCode(this.project.key, <Pipeline>this.dataToSave, this.selectedBranch, this.commitMessage)
-                    .subscribe(ope => {
-                    this.ope = ope;
-                    let zone = new NgZone({ enableLongStackTrace: false });
-                    let webworker = new CDSWebWorker('./assets/worker/web/operation.js');
-                    webworker.start({
-                        'user': this._store.selectSnapshot(AuthenticationState.user),
-                        'api': '/cdsapi',
-                        'path': '/project/' + this.project.key + '/workflows/' + this.name + '/ascode/' + this.ope.uuid
-                    });
-                    this.webworkerSub = webworker.response().subscribe(operation => {
-                        if (operation) {
-                            zone.run(() => {
-                                this.ope = JSON.parse(operation);
-                                if (this.ope.status > 1) {
-                                    this.loading = false;
-                                    webworker.stop();
-                                }
-                                this._cd.markForCheck();
-                            });
-                        }
+                this.loading = true;
+                this._pipService.updateAsCode(this.project.key, <Pipeline>this.dataToSave,
+                    this.selectedBranch, this.commitMessage).subscribe(o => {
+                        this.ope = o;
+                        this.startPollingOperation();
                     });
-                });
                 break;
             default:
                 this._toast.error('', this._translate.instant('ascode_error_unknown_type'))
         }
     }
+
+    startPollingOperation() {
+        this.pollingOperationSub = Observable.interval(1000).subscribe(() => {
+            this._workflowService.getAsCodeOperation(this.project.key, this.name, this.ope.uuid)
+                .pipe(finalize(() => this._cd.markForCheck()))
+                .subscribe(o => {
+                    this.ope = o;
+                    if (this.ope.status > 1) {
+                        this.loading = false;
+                        this.pollingOperationSub.unsubscribe();
```

Comment:
I Already have pb with calling unsubscribe in subscribe method.
I think, best way is doing something like that:  
.first(o => o.status > 1) .  and you will not need to take care of subscription

Reply:
Done using mergeMap and first